snuba/datasets/processors/replays_processor.py

```diff
@@ -211,6 +204,19 @@ def process_message(
 U = TypeVar("U")
 
 
+def segment_id_to_event_hash(segment_id: int | None) -> str:
+    if segment_id is None:
+        # Rows with null segment_id fields are considered "out of band" meaning they do not
+        # originate from the SDK and do not relate to a specific segment.
+        #
+        # For example: archive requests.
+        return str(uuid.uuid4().hex)
```

Comment:
It seems a little unusual to assign a random id to the message here. If somehow a message gets processed twice, the same message would have different ids. You lose out on any benefit of clickhouse deduplication. Is this acceptable? I'm wondering if it is better to do the id assignment from the source so once a message is created it has a unique id for life.

Reply:
@dbanda Updated to allow for setting event_hash explicitly.  Missing event hashes with null segment-id will still generate a random uuid.